### PR TITLE
DW-6985: Allows connection between analytical-env and AP frontend

### PR DESCRIPTION
**Pull request by nathanbigaignon**

### CODE REVIEW

File: src/main/kotlin/uk/gov/dwp/dataworks/services/TaskDeploymentService.kt [hunk 1]
```diff
@@ -214,7 +215,8 @@ class TaskDeploymentService {
             ".${configurationResolver.getStringConfig(ConfigKey.AWS_REGION)}.compute.internal",
             ".dataworks.dwp.gov.uk",
             "localhost",
-            "127.0.0.1"
+            "127.0.0.1",
+            ".${apFrontendUrl}"
```
Comment (connoravo) [inline src/main/kotlin/uk/gov/dwp/dataworks/services/TaskDeploymentService.kt:219]: You've already got a leading dot in the var.

File: terraform/modules/ecs-user/security_groups.tf [hunk 2]
```diff
@@ -155,3 +155,33 @@ resource "aws_security_group_rule" "ingress_emr_livy_from_host_hue" {
...
+  vpc_id      = var.vpc.id
+
+  ingress = [
+    {
+      description      = "Allow HTTPS ingress from orchestration-service-user-host SG"
+      from_port        = 443
+      to_port          = 443
+      protocol         = "tcp"
+      cidr_blocks      = []
+      ipv6_cidr_blocks = []
```
Comment (connoravo) [inline terraform/modules/ecs-user/security_groups.tf:171]: You are allowing ingress from nowhere

File: terraform/modules/ecs-user/security_groups.tf [hunk 3]
```diff
@@ -155,3 +155,33 @@ resource "aws_security_group_rule" "ingress_emr_livy_from_host_hue" {
...
+
+resource "aws_security_group_rule" "egress_to_ap_frontend_sg" {
+  description              = "Allow HTTPS egress to AP frontend VPCE"
+  from_port                = 443
+  to_port                  = 443
+  protocol                 = "tcp"
+  security_group_id        = aws_security_group.user_host.id
+  source_security_group_id = aws_security_group.ap_frontend_vpce_sg.id
+  type                     = "egress"
+}
```
Comment (connoravo) [inline terraform/modules/ecs-user/security_groups.tf:187]: New line at the end of all files please

File: src/main/kotlin/uk/gov/dwp/dataworks/services/TaskDeploymentService.kt [hunk 4]
```diff
@@ -203,6 +203,7 @@ class TaskDeploymentService {
         val hasFileTransferPermission = hasFileTransferDownloadPermission || hasFileTransferUploadPermission
         val hasClipboardOutPermission = authorizationService
             .hasUserToolingPermission(containerProperties.userName, ToolingPermission.CLIPBOARD_OUT)
+        val apFrontendUrl = System.getenv("AP_FRONTEND_VPCE").split(".").reversed().joinToString(".")
```
Comment (connoravo) [inline src/main/kotlin/uk/gov/dwp/dataworks/services/TaskDeploymentService.kt:206]:
```suggestion
        val apFrontendUrl = System.getenv("AP_FRONTEND_VPCE").split(".").reversed().joinToString(prefix = ".", separator = ".")
```

File: terraform/modules/ecs-user/vpce.tf [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+resource "aws_vpc_endpoint" "ap_frontend_vpce" {
+  vpc_id              = var.vpc.id
+  service_name        = var.ap_frontend_vpce
+  private_dns_enabled = true
+  security_group_ids  = [aws_security_group.ap_frontend_vpce_sg.id]
+  subnet_ids          = var.vpc.aws_subnets_private[*].id
+  vpc_endpoint_type   = "Interface"
+}
+
+resource "aws_route53_zone" "ap_zone" {
```
Comment (IanUtley) [inline terraform/modules/ecs-user/vpce.tf:11]: Why are we creating a route53 entry for AP? Shouldn't AP have setup the address in route53, it's an external environment to the Orchestration Service.

**Pull request closed without merging**